File: 2019/py/day2/day2.py
#!/usr/bin/env python3
"""Answer for Advent of Code 2019, Day 2.1."""


def part1():
    """Rebuild intcode from list."""
    numbers = read_input('./day2/input.txt')

    x = 0
    for num in numbers:
        if numbers[x] == 99:
            break

        a = numbers[x+1]
        b = numbers[x+2]
        c = numbers[x+3]

        if numbers[x] == 1:
            numbers[c] = numbers[a] + numbers[b]
        elif numbers[x] == 2:
            numbers[c] = numbers[a] * numbers[b]
        else:
            print('Something went wrong; got opcode {}'.format(numbers[x]))
        x += 4

    print(numbers[0])


def part2():
    """Do part two."""
    print("Part 2!")


def read_input(filename):
    """Parse input file and return list."""
    with open(filename, 'r') as file:
        for line in file:
            nums = line.split(',')
    x = 0
    for num in nums:
        nums[x] = int(num)
        x += 1

    nums[1] = 12
    nums[2] = 2
    return nums

File: 2019/php/day2/day2.php
<?php

class day2 {
  public function part1() {
    $nums = $this->run_intcode($this->read_input("./day2/input.txt"));
    echo $nums[0];
  }

  public function part2() {
  }

  private function read_input($file) {
    return explode(",", file_get_contents($file));
  }

  private function run_intcode($nums, $noun = 12, $verb = 2) {
    $x = 0;
    $nums[1] = $noun;
    $nums[2] = $verb;

    while($x < count($nums)) {
      if($nums[$x] == 99) {
        break;
      }

      $a = $nums[$x+1];
      $b = $nums[$x+2];
      $c = $nums[$x+3];

      if($nums[$x] == 1) {
        $nums[$c] = $nums[$a] + $nums[$b];
      } else if($nums[$x] == 2) {
        $nums[$c] = $nums[$a] * $nums[$b];
      } else {
        echo "Unexpected optcode: " . $num[$x];
      }
      $x += 4;
    }

    return $nums;
  }
}

?>